craft_application/util/yaml.py

```diff
@@ -97,6 +101,42 @@ def __init__(self, stream: TextIO) -> None:
         )
 
 
+class _SafeLineNoLoader(_SafeYamlLoader):
+    def __init__(self, stream: TextIO) -> None:
+        super().__init__(stream)
+
+        self.add_constructor(
+            yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, _dict_constructor
+        )
+
+    def compose_node(self, parent: Node | None, index: int) -> Node | None:
+        # the line number where the previous token has ended (plus empty lines)
+        line = self.line
+        node = Composer.compose_node(self, parent, index)
+        setattr(node, "__line__", line + 1)  # noqa: B010 - used internally, prevent mypy error
+        return node
+
+    def construct_mapping(
+        self,
+        node: MappingNode,
+        deep: bool = False,  # noqa: FBT001, FBT002 - used internally by yaml.SafeLoader
```

Comment:
I think you can eliminate that ruff linting error by putting an `@override` decorator on this method.